site/app/controllers/student/SubmissionController.php

```diff
@@ -259,6 +259,18 @@ private function uploadSubmission() {
                 return $this->uploadResult("File(s) uploaded too large.  Maximum size is ".($max_size/1000)." kb. Uploaded file(s) was ".($file_size/1000)." kb.", false);
             }
             
+            for($i = 1; $i <= $gradeable->getNumParts();$i++) {
+                if(isset($uploaded_files[$i])) {
+                    for($j = 0; $j < $count[$i]; $j++) {
+                        if($uploaded_files[$i]["is_zip"][$j] === true) {
+                            if(FileUtils::checkFileInZipName($uploaded_files[$i]["tmp_name"][$j]) === false) {
+                                return $this->uploadResult("You may not use quotes, backslashes or angle brackets in your filename for files inside the zip.", false);
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
You should put the inner most if statement into [this if statement right above](https://github.com/Submitty/Submitty/blob/44157f635c412d3ff7b3d88f2068300ac22a1829/site/app/controllers/student/SubmissionController.php#L241). You would then also have a check on files that aren't zips in the else clause.